src/xenia/kernel/xthread.cc

```diff
@@ -709,6 +761,376 @@ X_STATUS XThread::Delay(uint32_t processor_mode, uint32_t alertable,
   }
 }
 
+bool XThread::StepToAddress(uint32_t pc) {
+  auto functions = emulator()->processor()->FindFunctionsWithAddress(pc);
+  if (functions.empty()) {
+    // Function hasn't been generated yet. Generate it.
+    if (!emulator()->processor()->ResolveFunction(pc)) {
+      XELOGE("XThread::StepToAddress(%.8X) - Function could not be resolved",
+             pc);
+      return false;
+    }
+  }
+
+  // Instruct the thread to step forwards.
+  threading::Fence fence;
+  cpu::Breakpoint bp(kernel_state()->processor(), pc,
+                     [&fence](uint32_t guest_address, uint64_t host_address) {
+                       fence.Signal();
+                     });
+  if (bp.Install()) {
+    thread_->Resume();
+    fence.Wait();
+    bp.Uninstall();
+  } else {
+    assert_always();
+    XELOGE("XThread: Could not install breakpoint to step forward!");
+    return false;
+  }
+
+  return true;
+}
+
+uint32_t XThread::StepIntoBranch(uint32_t pc) {
+  cpu::frontend::InstrData i;
+  i.address = pc;
+  i.code = xe::load_and_swap<uint32_t>(memory()->TranslateVirtual(i.address));
+  i.type = cpu::frontend::GetInstrType(i.code);
+
+  auto context = thread_state_->context();
+
+  if (i.type->type & (1 << 4) /* BranchAlways */) {
```

Comment:
Did that cause (for example) the BranchAlways enum matches BranchAlways and Branch - basically matching all branching instructions.